File: java/src/main/java/challenges/aoc2023/day2/Application.java
package challenges.aoc2023.day2;

import challenges.aoc2023.day2.service.DayTwoSolver;
import service.TaskSolver;

public class Application {
    public static void main(String[] args) {
        final TaskSolver solver = new TaskSolver(new DayTwoSolver());
        solver.solveTasks();
    }
}

File: scala/src/main/scala/challenges/aoc2025/day2/Application.scala
package challenges.aoc2025.day2

import _root_.service.TaskSolver
import challenges.aoc2025.day2.service.{DayTwoSolver, IdValidationService}

object Application {
  def main(args: Array[String]): Unit = {
    val solver: TaskSolver = TaskSolver(DayTwoSolver(IdValidationService()))
    solver.solveTasks()
  }
}
